### PR TITLE
ITelegramBotClient

### PR DESCRIPTION
**Pull request by MrZoidberg**
Interface for TelegramBotClient that allows easy mocking and faking of TelegramBotClient.

### CODE REVIEW

**Comment by MrRoundRobin** [comment]
how to deal with deprecated methods? is it best practice to include these method or should these methods be removed directly?

**Comment by thoemmi** [comment]
If someone's using the new interface, he's changing his code anyway. Therefore I wouldn't include the obsolete methods in the interface.

File: src/Telegram.Bot/ITelegramBotClient.cs [hunk 1]
```diff
@@ -0,0 +1,1133 @@
...
+
+        /// <summary>
+        /// Occurs when an <see cref="Update"/> is received.
+        /// </summary>
+        event EventHandler<UpdateEventArgs> OnUpdate;
+
+        /// <summary>
+        /// Occurs when an <see cref="Update"/> is received.
+        /// </summary>
+        event EventHandler<UpdateEventArgs> UpdateReceived;
```
Comment (MrRoundRobin) [inline src/Telegram.Bot/ITelegramBotClient.cs]: Can you remove the Obsolete events, properties/fields and methods?

**Comment by MrZoidberg** [comment]
@MrRoundRobin Done. Somewhat a mess with commits :(

**Comment by MrRoundRobin** [comment]
thx

**Pull request merged**